lib/util/template/index.ts

```diff
@@ -201,15 +207,23 @@ export function compile(
   input: CompileInput,
   filterFields = true
 ): string {
-  const data = { ...GlobalConfig.get(), ...input };
+  const data = merge.all([
+    GlobalConfig.get(),
+    { env: getChildEnv({}) },
+    input ?? {},
+  ]);
```

Comment:
don't merge! fix tests instead. input should never have an `env` key

Reply:
Mocking! 🤦

Done.

---

I'm not 100% happy about my implementation, because I don't know how to mock `getChildEnv` selectively, so that some tests use the original implementation and some the mocked one.

I removed those of my new unit tests that required the original `getChildEnv`.